grpc/interceptors.go

```diff
@@ -177,9 +179,28 @@ func (ci *clientInterceptor) intercept(
 	// And defer decrementing it when we're done
 	defer ci.metrics.inFlightRPCs.With(labels).Dec()
 	// Handle the RPC
+	begin := time.Now()
 	err := ci.metrics.grpcMetrics.UnaryClientInterceptor()(localCtx, fullMethod, req, reply, cc, invoker, opts...)
 	if err != nil {
 		err = unwrapError(err, respMD)
 	}
+	if status.Code(err) == codes.DeadlineExceeded {
+		return deadlineDetails{
+			service: service,
+			method:  method,
+			latency: time.Since(begin),
+		}
+	}
 	return err
 }
+
+type deadlineDetails struct {
```

Comment:
Can you add a Godoc comment here briefly describing this type and why it exists?